apps/open-swe/src/tools/builtin-tools/view.ts

```diff
@@ -1,33 +1,77 @@
 import { tool } from "@langchain/core/tools";
-import { GraphState } from "@open-swe/shared/open-swe/types";
+import { GraphState, GraphConfig } from "@open-swe/shared/open-swe/types";
 import { createLogger, LogLevel } from "../../utils/logger.js";
 import { getRepoAbsolutePath } from "@open-swe/shared/git";
 import { getSandboxSessionOrThrow } from "../utils/get-sandbox-id.js";
 import { createViewToolFields } from "@open-swe/shared/open-swe/tools";
 import { handleViewCommand } from "./handlers.js";
+import {
+  isLocalMode,
+  getLocalWorkingDirectory,
+} from "../../utils/local-mode.js";
+import { getLocalShellExecutor } from "../../utils/local-shell-executor.js";
+import { TIMEOUT_SEC } from "@open-swe/shared/constants";
 
 const logger = createLogger(LogLevel.INFO, "ViewTool");
 
 export function createViewTool(
   state: Pick<GraphState, "sandboxSessionId" | "targetRepository">,
+  config: GraphConfig,
 ) {
   const viewTool = tool(
     async (input): Promise<{ result: string; status: "success" | "error" }> => {
       try {
-        const sandbox = await getSandboxSessionOrThrow(input);
-        const workDir = getRepoAbsolutePath(state.targetRepository);
-
-        const { command, path, view_range } = input;
+        const { command, path, view_range } = input as any;
         if (command !== "view") {
           throw new Error(`Unknown command: ${command}`);
         }
 
-        const result = await handleViewCommand(
-          sandbox,
-          path,
-          workDir,
-          view_range as [number, number] | undefined,
-        );
+        let workDir;
```

Comment:
always prefer typing these vars:
```suggestion
        let workDir: string;
```

Reply:
same w/ occurrences below/elsewhere in this PR